### DISCUSSION TITLE
What is the minimum ImGui version required?

### DISCUSSION THREAD

**Discussion by JamAndCheese** [epezent/implot#212] [Q&A]
My application uses ImGui 1.67. I am happy with it, so have not upgraded in a while. I am trying to use ImPlot today but my compilation is failing with  "error: ‘ImGuiMouseButton’ has not been declared" . Also ImGuiKeyModFlags and others. I think my ImGui version might be too old. 

I would like to know the minimum version so I can put some calories in upgrading my app, but I am not ready to do a wholesale revamp (hence the 'minimum' question).

Thanks for any info!

**Comment by epezent** [comment]
Quite frankly, I do not know and have not spent the time to find out. ImPlot development began on ImGui 1.75 (almost exactly 1 year ago), but compatibility may potentially go further back than that (but apparently not to 1.67 in your case). It would be worthwhile to to iteratively test past releases of ImGui (at the very least for compilation issues), and stick a warning in our README.

**Reply by JamAndCheese** [reply]
Well, in case it helps anybody else: I upgraded my application to ImGui 1.82 and recorded all the API and build changes necessary. It was not too difficult.

**Reply by joenot443** [reply]
What were the necessary changes?

**Reply by JamAndCheese** [reply]
Oh wow. I did it a year ago based on imgui-0.10. As I remember, there was an ImGui API change dealing with mouse position reading, and some changes I had to do to be compatible with C++11  struct initialization list.
By now all these are all probably obsolete. If you really need me, I can find out, but you are probably better just using the latest ImPlot and ImGui.
